src/tests/ftest/util/nvme_utils.py

```diff
@@ -0,0 +1,451 @@
+from mpio_utils import MpioUtils
+
+try:
+    # python 3.x
+    import queue
+except ImportError:
+    # python 2.7
+    import Queue as queue
+
+def get_device_ids(dmg, servers):
+    """Get the NVMe Device ID from servers
+
+    Args:
+        dmg: DmgCommand class instance.
+        servers (list): list of server hosts.
+
+    Returns:
+        devices (dictionary): Device UUID for servers.
+
+    """
+    devices = {}
+    dmg.set_sub_command("storage")
+    dmg.sub_command_class.set_sub_command("query")
+    dmg.sub_command_class.sub_command_class.set_sub_command("list-devices")
+    for host in servers:
+        dmg.hostlist = host
+        try:
+            result = dmg.run()
+        except CommandFailure as _error:
+            raise "dmg command failed for list-devices"
+        drive_list = []
+        for line in result.stdout.split('\n'):
+            if 'UUID' in line:
+                drive_list.append(line.split(':')[1])
+        devices[host] = drive_list
+    return devices
+
+class ServerFillUp(IorTestBase):
+    """
+    Class to fill up the servers based on pool percentage given.
+    It will get the drives listed in yaml file and find the maximum capacity of
+    the pool which will be created.
+    IOR block size will be calculated as part of function based on percentage
+    of pool needs to fill up.
+    """
+    # pylint: disable=too-many-ancestors
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, *args, **kwargs):
+        """Initialize a IorTestBase object."""
+        super(ServerFillUp, self).__init__(*args, **kwargs)
+        self.no_of_pools = 1
+        self.capacity = 1
+        self.no_of_servers = 1
+        self.no_of_drives = 1
+        self.pool = None
+        self.dmg = None
+        self.container_info = {}
+        self.set_faulty_device = False
+
+    def setUp(self):
+        """Set up each test case."""
+        # obtain separate logs
+        self.update_log_file_names()
+        # Start the servers and agents
+        super(ServerFillUp, self).setUp()
+        # Recreate the client hostfile without slots defined
+        self.hostfile_clients = write_host_file(
+            self.hostlist_clients, self.workdir, None)
+        #Get this IOR Parameter from yaml file
+        self.ior_flags = self.params.get("ior_flags",
+                                         '/run/ior/iorflags/*')
+        self.ior_read_flags = self.params.get("ior_read_flags",
+                                              '/run/ior/iorflags/*', '-r -R')
+        self.ior_apis = self.params.get("ior_api", '/run/ior/iorflags/*')
+        self.ior_transfer_size = str(self.params.get("transfer_size",
+                                                     '/run/ior/iorflags/*'))
+        self.ior_daos_oclass = self.params.get("obj_class",
+                                               '/run/ior/iorflags/*', "SX")
+        self.processes = self.params.get("slots", "/run/ior/clientslots/*")
```

Comment:
Any specific reason these are being override and not using the original yaml file locations inherited by IorTestBase?